cpp/farm_ng/core/prototools/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+#[[
+sophus_serial
+
+
+
+]]
+
+set(farm_ng_core_prototools_src_prefixes
+  proto_reader_writer
+)
+
+set(farm_ng_core_prototools_cpp)
+set(farm_ng_core_prototools_h)
+foreach(x ${farm_ng_core_prototools_src_prefixes} )
+    list(APPEND farm_ng_core_prototools_cpp ${x}.cpp)
+    list(APPEND farm_ng_core_prototools_h ${x}.h)
+endforeach()
+
+add_library(farm_ng_core_prototools SHARED
+  ${farm_ng_core_prototools_cpp}
+  ${farm_ng_core_prototools_h})
+target_link_libraries(farm_ng_core_prototools protobuf::libprotobuf farm_ng_core_logging)
+add_library (farm_ng_core::farm_ng_core_prototools ALIAS farm_ng_core_prototools)
+target_include_directories(farm_ng_core_prototools PUBLIC
```

Comment:
I would recommend making a cmake macro that automates these patterns if possible.

Reply:
yes, will do for sure. It is planned.